sdk/src/types/block/output/feature/metadata.rs

```diff
@@ -27,77 +59,108 @@ impl MetadataFeature {
 
     /// Creates a new [`MetadataFeature`].
     #[inline(always)]
-    pub fn new(data: impl Into<Vec<u8>>) -> Result<Self, Error> {
-        Self::try_from(data.into())
+    pub fn new(data: impl IntoIterator<Item = (Vec<u8>, Vec<u8>)>) -> Result<Self, Error> {
+        let data: BTreeMap<Vec<u8>, Vec<u8>> = data.into_iter().collect();
+        let metadata = Self::try_from(data)?;
+
+        verify_keys_packable::<true>(&metadata.0)?;
+
+        Ok(metadata)
     }
 
     /// Returns the data.
     #[inline(always)]
-    pub fn data(&self) -> &[u8] {
-        &self.0
+    pub fn data(
+        &self,
+    ) -> &BTreeMap<BoxedSlicePrefix<u8, MetadataFeatureKeyLength>, BoxedSlicePrefix<u8, MetadataFeatureValueLength>>
+    {
+        self.0.deref()
     }
 }
 
 impl StorageScore for MetadataFeature {}
 
 impl WorkScore for MetadataFeature {}
 
-macro_rules! impl_from_vec {
-    ($type:ty) => {
-        impl TryFrom<$type> for MetadataFeature {
-            type Error = Error;
+impl TryFrom<Vec<(Vec<u8>, Vec<u8>)>> for MetadataFeature {
+    type Error = Error;
 
-            fn try_from(value: $type) -> Result<Self, Self::Error> {
-                Vec::<u8>::from(value).try_into()
+    fn try_from(data: Vec<(Vec<u8>, Vec<u8>)>) -> Result<Self, Error> {
```

Comment:
Maybe there is a way to use the other try_from inside to avoid code duplication

Reply:
Just moved it to a new function that accepts IntoIterator